### ISSUE TITLE
Plate #7443 breaks Marvin API? (http 500)

### ISSUE THREAD

**Issue by ericliang45** [sdss/marvin#129]
When I ran the online example of API using the plate #7443 (and only this plate), I got an HTTP Status Error: 500. Maybe as a commisioning plate, it has something special that breaks Marvin.

The example/code I used:
```
from marvin import config
config.mode = 'remote'
from marvin.api.api import Interaction
plateifu = '7443-12701'
url = config.urlmap['api']['getCube']['url']
response = Interaction(url.format(name=plateifu))
```
The Error message:
<img width="1103" alt="marvin-api http 500" src="https://cloud.githubusercontent.com/assets/23513046/21211175/5017e67c-c2bc-11e6-89a4-d9e5af7a2c78.png">

**Comment by havok2063** [comment]
If this a bug, bumping this to 2.1.  I think we found the problem back in December.

**Comment by havok2063** [comment]
Here is the output of the log from this error
```
2017-01-27 10:16:10 - ERROR - Exception on /marvin2/api/cubes/7443-12701/ [POST] [log_exception @ app.py]
Traceback (most recent call last):
  File "/uufs/chpc.utah.edu/sys/pkg/python/2.7.3_rhel6/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/uufs/chpc.utah.edu/sys/pkg/python/2.7.3_rhel6/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/uufs/chpc.utah.edu/sys/pkg/python/2.7.3_rhel6/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/uufs/chpc.utah.edu/sys/pkg/python/2.7.3_rhel6/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/uufs/chpc.utah.edu/sys/pkg/python/2.7.3_rhel6/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/uufs/chpc.utah.edu/sys/pkg/python/2.7.3_rhel6/lib/python2.7/site-packages/flask_profiler-1.0.1-py2.7.egg/flask_profiler/flask_profiler.py", line 111, in wrapper
    return wrapped(*args, **kwargs)
  File "/uufs/chpc.utah.edu/sys/pkg/python/2.7.3_rhel6/lib/python2.7/site-packages/flask_profiler-1.0.1-py2.7.egg/flask_profiler/flask_profiler.py", line 83, in wrapper
    returnVal = f(*args, **kwargs)
  File "/uufs/chpc.utah.edu/sys/pkg/python/2.7.3_rhel6/lib/python2.7/site-packages/Flask_Classy-0.6.10-py2.7.egg/flask_classy.py", line 200, in proxy
    response = view(**request.view_args)
  File "/home/manga/software/git/manga/marvin/2.0.9/python/marvin/api/cube.py", line 74, in get
    'redshift': cube.data.target.NSA_objects[0].z,
AttributeError: 'NoneType' object has no attribute 'NSA_objects'
```

Now that the new code in api/cube.py has changed to `'redshift': cube.nsa.z`, this problem should be fixed.  After we merge and assess at Utah, I think we can close this.

**Comment by havok2063** [comment]
So I tried this locally as a test and got this error.  How does the new NSA code handle cubes that don't have NSA data available? @albireox 

```
inside global session vers v2_0_1 2.0.2
my form {'release': u'MPL-4'}
WARNING: no NSA targets found for this cube.
Traceback (most recent call last):
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask/app.py", line 1994, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask_profiler/flask_profiler.py", line 113, in wrapper
    return wrapped(*args, **kwargs)
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask_profiler/flask_profiler.py", line 85, in wrapper
    returnVal = f(*args, **kwargs)
  File "/Users/Brian/anaconda2/lib/python2.7/site-packages/flask_classy.py", line 200, in proxy
    response = view(**request.view_args)
  File "/Users/Brian/Work/github_projects/Marvin/python/marvin/api/cube.py", line 173, in get
    'redshift': cube.nsa.z,
  File "/Users/Brian/Work/github_projects/Marvin/python/marvin/core/core.py", line 281, in nsa
    drpall=self._drpall)
  File "/Users/Brian/Work/github_projects/Marvin/python/marvin/utils/general/general.py", line 887, in get_nsa_data
    drpver=drpver, drpall=drpall)
  File "/Users/Brian/Work/github_projects/Marvin/python/marvin/utils/general/general.py", line 950, in get_nsa_data
    url = marvin.config.urlmap['api'][request_name]['url']
  File "/Users/Brian/Work/github_projects/marvin_brain/python/brain/core/core.py", line 39, in __missing__
    raise BrainError('Key {0} not found in urlmap.'.format(key))
BrainError: Key nsa_full not found in urlmap.
```

**Comment by albireox** [comment]
This should be fixed as for #169
